diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7ebcf4e..9766201 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -226,7 +226,7 @@ if(ENABLE_PYTHON)
 endif(ENABLE_PYTHON)
 
 # install library
-install(TARGETS netconf2 DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS netconf2 DESTINATION ${CMAKE_INSTALL_LIBDIR} libnetconf2)
 
 # install headers
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/nc_client.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
@@ -277,3 +277,6 @@ add_custom_target(cleancache
                   COMMAND find . -iname '*cmake*' -not -name CMakeLists.txt -exec rm -rf {} +
                   COMMAND rm -rf Makefile Doxyfile
                   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
++include(PackagingLibNetconf)
++include(Packaging)
++make_packages()
