diff --git a/cli/CMakeLists.txt b/cli/CMakeLists.txt
index cc20d17..12408e1 100644
--- a/cli/CMakeLists.txt
+++ b/cli/CMakeLists.txt
@@ -65,11 +65,11 @@ include(CheckFunctionExists)
 check_function_exists(eaccess HAVE_EACCESS)
 check_function_exists(mkstemps HAVE_MKSTEMPS)
 if(HAVE_MKSTEMPS)
-    set(CMAKE_C_FLAGS         "${CMAKE_C_FLAGS} -DHAVE_MKSTEMPS")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DHAVE_MKSTEMPS")
 endif(HAVE_MKSTEMPS)
 
 # install binary
-install(TARGETS netopeer2-cli DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(TARGETS netopeer2-cli DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT netopeer2)
 
 # man -> html
 find_program(ROFF2HTML roff2html)
@@ -81,7 +81,7 @@ else()
 endif()
 
 # install doc (man)
-install(FILES ${PROJECT_SOURCE_DIR}/doc/${PROJECT_NAME}.1 DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
+install(FILES ${PROJECT_SOURCE_DIR}/doc/${PROJECT_NAME}.1 DESTINATION ${CMAKE_INSTALL_MANDIR}/man1 COMPONENT netopeer2)
 
 # clean cmake cache
 add_custom_target(cleancache
@@ -92,3 +92,7 @@ add_custom_target(cleancache
 
 # uninstall
 add_custom_target(uninstall "${CMAKE_COMMAND}" -P "${CMAKE_MODULE_PATH}/uninstall.cmake")
+
+include(PackagingNetopeer)
+include(Packaging)
+make_packages()
diff --git a/server/CMakeLists.txt b/server/CMakeLists.txt
index 93b5e3c..8f8918f 100755
--- a/server/CMakeLists.txt
+++ b/server/CMakeLists.txt
@@ -172,7 +172,7 @@ if(NP2SRV_ENABLED_URL_CAPABILITY)
 endif()
 
 # install binary
-install(TARGETS netopeer2-server DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(TARGETS netopeer2-server DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT netopeer2)
 
 # only for configuration
 if (ENABLE_CONFIGURATION)
@@ -205,7 +205,7 @@ if (ENABLE_CONFIGURATION)
         set(ENV{SYSREPOCTL} ${SYSREPOCTL_EXECUTABLE})
         set(ENV{SYSREPOCFG} ${SYSREPOCFG_EXECUTABLE})
         set(ENV{SYSREPOCTL_ROOT_PERMS} SYSREPOCTL_ROOT_PERMS)
-        execute_process(COMMAND ${PROJECT_BINARY_DIR}/model-install.sh)")
+        execute_process(COMMAND ${PROJECT_BINARY_DIR}/model-install.sh)" COMPONENT netopeer2)
 endif()
 
 add_custom_target(install-scripts-ide
@@ -244,3 +244,7 @@ add_custom_target(integration-test
                   COMMENT Running integration-tests)
 
 file(COPY integration-tests DESTINATION ${CMAKE_BINARY_DIR})
+
+include(PackagingNetopeer)
+include(Packaging)
+make_packages()
