diff --git a/cli/CMakeLists.txt b/cli/CMakeLists.txt
index fcd9f1b..d66ac76 100644
--- a/cli/CMakeLists.txt
+++ b/cli/CMakeLists.txt
@@ -69,7 +69,7 @@ if(HAVE_MKSTEMPS)
 endif(HAVE_MKSTEMPS)
 
 # install binary
-install(TARGETS netopeer2-cli DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(TARGETS netopeer2-cli DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT netopeer2)
 
 # man -> html
 find_program(ROFF2HTML roff2html)
diff --git a/keystored/CMakeLists.txt b/keystored/CMakeLists.txt
index 33aec1e..ea0ac86 100644
--- a/keystored/CMakeLists.txt
+++ b/keystored/CMakeLists.txt
@@ -153,4 +153,4 @@ add_custom_target(install-scripts-ide
 )
 
 # plugins should be installed into sysrepo plugins dir
-install(TARGETS keystored DESTINATION ${SR_PLUGINS_DIR})
+install(TARGETS keystored DESTINATION ${SR_PLUGINS_DIR} COMPONENT netopeer2)
diff --git a/server/CMakeLists.txt b/server/CMakeLists.txt
index be34b15..3799791 100755
--- a/server/CMakeLists.txt
+++ b/server/CMakeLists.txt
@@ -168,7 +168,7 @@ if(NP2SRV_ENABLED_URL_CAPABILITY)
 endif()
 
 # install binary
-install(TARGETS netopeer2-server DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(TARGETS netopeer2-server DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT netopeer2)
 
 # only for configuration
 if (ENABLE_CONFIGURATION)
