diff --git a/CMakeLists.txt b/CMakeLists.txt
index b50df764..c0577bcf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -355,3 +355,6 @@ if(NOT CALL_TARGET_BINS_DIRECTLY)
     install(CODE "file(WRITE \"${install_yang_file_name}\" \"${SH_INSTALL_YANG_CMDS}\")")
     install(CODE "message(STATUS \"Installation commands were written to ${install_yang_file_name}\")")
 endif()
+include(PackagingSysRepo)
+include(Packaging)
+make_packages()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5aba8463..4e23d66c 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -122,7 +122,7 @@ target_link_libraries(sysrepoctl sysrepo ${YANG_LIBRARIES})
 add_executable(sysrepocfg ${EXECUTABLES_DIR}/sysrepocfg.c)
 target_link_libraries(sysrepocfg sysrepo ${YANG_LIBRARIES})
 
-install(TARGETS sysrepo DESTINATION ${CMAKE_INSTALL_LIBDIR})
-install(TARGETS sysrepo_a DESTINATION ${CMAKE_INSTALL_LIBDIR})
-install(TARGETS sysrepod sysrepo-plugind sysrepoctl sysrepocfg DESTINATION ${CMAKE_INSTALL_BINDIR})
-install(DIRECTORY "${PROJECT_SOURCE_DIR}/inc/" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(TARGETS sysrepo DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT sysrepo)
+install(TARGETS sysrepo_a DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT sysrepo)
+install(TARGETS sysrepod sysrepo-plugind sysrepoctl sysrepocfg DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT sysrepo)
+install(DIRECTORY "${PROJECT_SOURCE_DIR}/inc/" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} COMPONENT sysrepo)
